WordPress/Classes/Utility/BuildInformation/Build.swift

```diff
@@ -0,0 +1,73 @@
+@objc
+class Build: NSObject {
```

Comment:
Is this used in ObjC anywhere?

Reply:
It's not, although I ended up putting a lot of effort into making the code Obj-C compatible since we have so much Obj-C code in our app, and our unit tests could end up needing it.

This is one of the reasons why I think we should make it a priority to migrate our code to Swift.

At the same time I'd hate this code being duplicated just because it didn't work for Obj-C (if that makes any sense).

Reply:
The original reason (maybe not a great one) for keeping `Build` inside the `FeatureFlag.swift` was that it was meant as a helper for the feature flags. As you mentioned somewhere else, the other types can be considered _code smells_, and I'd rather not have anything other than `Feature` check for the build configuration.

Reply:
That sounds about right, but that will affect how we test that class.  Overriding the value of a private detail should not be done by automated tests.

Reply:
Wouldn't [this](https://github.com/wordpress-mobile/WordPress-iOS/pull/7896#issuecomment-334075022) be enough for testing?

Reply:
I'm not saying that's not enough... but [these](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/WordPressTest/FeatureFlagTest.swift#L8) tests involve manipulating `Build` information.

The approach used in those tests to override the `Build` information is one that I particularly don't like.  In fact for cases like that I find that dependency injection is a much more appropriate approach.  But dependency injection would require `Build` not to be a private class.